thoth/build_analysers/cli.py

```diff
@@ -82,71 +93,151 @@ def cli():
 
 
 @cli.command()
-@click.argument("log", type=click.Path(exists=True))
-@click.option("--limit", "-n", help="Limit number of candidates.", type=int)
+@click.argument("log", envvar="THOTH_BUILD_ANALYSER_LOG_PATH", type=click.Path())
+@click.option("--limit", "-n", envvar="THOTH_BUILD_ANALYSER_LIMIT", help="Limit number of candidates.", type=int)
 @click.option(
     "--handler", help="Handler to parse log dependencies.", type=click.Choice(choices=["pip3", "pipenv"]), default=None
 )
+@click.option(
+    "--ceph-document-id",
+    help="Document id of buil log in ceph",
+    envvar="THOTH_BUILD_LOG_DOC_ID",
+    default="-",
+    type=str,
+    required=False,
+)
 @click.option("--colorize/--no-colorize", default=True)
 @click.option("--pretty", "-p", is_flag=True, default=False)
 def report(
-    log: Union[str, Path], limit: int = 5, handler: str = None, colorize: bool = False, pretty: bool = False
+    log: str,
+    ceph_document_id: str = "-",
+    limit: int = 5,
+    handler: str = None,
+    colorize: bool = False,
+    pretty: bool = False,
 ) -> str:
     """Analyze raw build log and produce a report."""
-    with open(log, "r") as f:
-        build_log: str = f.read()
+    if ceph_document_id is not "-":
+        _get_document(ceph_document_id, log)
 
-    result: dict = build_breaker_report(log=build_log, handler=handler, top=limit, colorize=colorize)
+    with open(log, "r") as f:
+        build_log_detail: json = json.load(f)
 
-    if pretty:
-        result: str = pformat(result)
+    result: dict = build_breaker_report(log=build_log_detail["log"], handler=handler, top=limit, colorize=colorize)
 
-        click.echo(result)
-        sys.exit(0)
+    if ceph_document_id is "-":
```

Comment:
If `ceph_document_id` is not provided (its not required)? I would remove the special case of `-` (remove that default) and just check if was set:

```python
    if ceph_document_id:
```

Here and bellow.